src/backend/distributed/commands/statistics.c

```diff
@@ -0,0 +1,206 @@
+
+
+/*
+ * PreprocessCreateStatisticsStmt is called during the planning phase for
+ * CREATE STATISTICS.
+ */
+List *
+PreprocessCreateStatisticsStmt(Node *node, const char *queryString)
+{
+	CreateStatsStmt *stmt = castNode(CreateStatsStmt, node);
+
+	RangeVar *relation = (RangeVar *) linitial(stmt->relations);
+	Oid relationId = RangeVarGetRelid(relation, ShareUpdateExclusiveLock, false);
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	QualifyTreeNode((Node *) stmt);
+
+	char *ddlCommand = DeparseTreeNode((Node *) stmt);
+
+	DDLJob *ddlJob = palloc0(sizeof(DDLJob));
+
+	ddlJob->targetRelationId = RangeVarGetRelid(relation, ShareUpdateExclusiveLock,
+												false);
+	ddlJob->concurrentIndexCmd = false;
+	ddlJob->startNewTransaction = false;
+	ddlJob->commandString = ddlCommand;
+	ddlJob->taskList = DDLTaskList(relationId, ddlCommand);
+
+	List *ddlJobs = list_make1(ddlJob);
+
+	return ddlJobs;
+}
+
+
+/*
+ * PostprocessCreateStatisticsStmt is called after a CREATE STATISTICS command has
+ * been executed by standard process utility.
+ */
+List *
+PostprocessCreateStatisticsStmt(Node *node, const char *queryString)
+{
+	CreateStatsStmt *stmt = castNode(CreateStatsStmt, node);
+	Assert(stmt->type == T_CreateStatsStmt);
+
+	RangeVar *relation = (RangeVar *) linitial(stmt->relations);
+	Oid relationId = RangeVarGetRelid(relation, ShareUpdateExclusiveLock, false);
+
+	if (!IsCitusTable(relationId) || !ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	bool missingOk = false;
+	ObjectAddress objectAddress = GetObjectAddressFromParseTree((Node *) stmt, missingOk);
+
+	EnsureDependenciesExistOnAllNodes(&objectAddress);
+
+	return NIL;
+}
+
+
+/*
+ * CreateStatisticsStmtObjectAddress finds the ObjectAddress for the statistics that
+ * is created by given CreateStatsStmt. If missingOk is false and if statistics
+ * does not exist, then it errors out.
+ *
+ * Never returns NULL, but the objid in the address can be invalid if missingOk
+ * was set to true.
+ */
+ObjectAddress
+CreateStatisticsStmtObjectAddress(Node *node, bool missingOk)
+{
+	CreateStatsStmt *stmt = castNode(CreateStatsStmt, node);
+
+	ObjectAddress address = { 0 };
+	Oid statsOid = get_statistics_object_oid(stmt->defnames, missingOk);
+	ObjectAddressSet(address, StatisticExtRelationId, statsOid);
+
+	return address;
+}
+
+
+/*
+ * GetExplicitStatisticsCommandList returns the list of DDL commands to create
+ * statistics that are explicitly created for the table with relationId. See
+ * comment of GetExplicitStatisticsIdList function.
+ */
+List *
+GetExplicitStatisticsCommandList(Oid relationId)
+{
+	List *createStatisticsCommandList = NIL;
+
+	PushOverrideEmptySearchPath(CurrentMemoryContext);
+
+	List *statisticsIdList = GetExplicitStatisticsIdList(relationId);
+
+	Oid statisticsId = InvalidOid;
+	foreach_oid(statisticsId, statisticsIdList)
+	{
+		char *createStatisticsCommand = pg_get_statisticsobj_worker(statisticsId, true);
```

Comment:
If missing_ok is true we might need to check if the result is NULL or not. (or maybe missing_ok needs to be false)

Reply:
I think it should be false